app/src/main/java/com/android/streetworkapp/ui/image/AddImageButton.kt

```diff
@@ -0,0 +1,134 @@
+package com.android.streetworkapp.ui.image
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.net.Uri
+import android.util.Log
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import com.android.sample.R
+import com.android.streetworkapp.model.park.Park
+import com.android.streetworkapp.ui.theme.ColorPalette
+import java.io.File
+
+object AddImageButtonParams {
+  const val DEBUG_PREFIX = "AddImageButton:"
+}
+
+/**
+ * IconButton that on onclick will request permission (if not already done) and open the camera to
+ * take a picture. After a picture is taken, a [ConfirmImageDialog] will be shown to prompt the user
+ * to cancel or upload the file
+ *
+ * @param currentPark The info of the park the image will be linked to
+ * @param context The current context
+ */
+@Composable
+fun AddImageButton(currentPark: Park?) {
+
+  val context = LocalContext.current
+
+  var hasCameraPermission by remember {
+    mutableStateOf(
+        ContextCompat.checkSelfPermission(context, Manifest.permission.CAMERA) ==
+            PackageManager.PERMISSION_GRANTED)
+  }
+
+  var showConfirmationDialog by remember { mutableStateOf(false) }
+  var capturedImageUri by remember { mutableStateOf<Uri?>(null) }
+
+  // Temporary file for storing the captured image
+  val tempFile by remember {
+    mutableStateOf(
+        File(context.cacheDir, "temp_image_for_park_upload.jpg").apply {
+          if (exists()) {
```

Comment:
Sugestion: those "if" perhaps be used in the same line , using a "&&" operator:

if(exists() && !delete()) 

(I might be missing something, if there is a reason for this nested if, perhaps we could add a small comment in the code)

Reply:
done